### ISSUE TITLE
Work with Hyperledger Fabric Chaincode Development Tools

### ISSUE THREAD

**Issue by ffffffffffk** [SAPDocuments/Tutorials#4415]
Tutorial URL: https://developers.sap.com/tutorials/blockchain-hlf-chaincode-tools.html

Please specify the step you are referring to

I have encountered the following errors in the tutorial:

C:\Users\xxxxxx>go get github.com/hyperledger/fabric/core/chaincode/shim
package github.com/hyperledger/fabric/core/chaincode/shim: cannot find package "github.com/hyperledger/fabric/core/chaincode/shim" in any of:
        c:\go\src\github.com\hyperledger\fabric\core\chaincode\shim (from $GOROOT)
        C:\Users\i350156\go\src\github.com\hyperledger\fabric\core\chaincode\shim (from $GOPATH)

C:\Users\xxxxxx>go get github.com/hyperledger/fabric/protos/peer
package github.com/hyperledger/fabric/protos/peer: cannot find package "github.com/hyperledger/fabric/protos/peer" in any of:
        c:\go\src\github.com\hyperledger\fabric\protos\peer (from $GOROOT)
        C:\Users\i350156\go\src\github.com\hyperledger\fabric\protos\peer (from $GOPATH)

**Comment by heytombeck** [comment]
Hi - I've just looked at the github for Hyperledger Fabric and it appears that they've changed the structure / what's available. I've raised this with one of our developers and I'll see if there's an alternative.

**Comment by heytombeck** [comment]
Unfortunately all three of my Hyperledger Fabric developers are showing an out of office until January 2nd, so it might take some time. While waiting, you can access the full HLF github profile here: https://github.com/hyperledger/fabric

**Comment by heytombeck** [comment]
Hello - I've heard back from the developers now and we'll edit the tutorial to fit around the new structure. The instructions will then be:

**Initialize GO Modules**
go mod init chaincode

**Install Hyperledger Fabric packages**
go get github.com/hyperledger/fabric/common/util@v1.4

go get github.com/hyperledger/fabric/core/chaincode/shim@v1.4

go get github.com/hyperledger/fabric/protos/peer@v1.4

**Comment by thecodester** [comment]
Thank you for your feedback. The issue seems to be resolved, so I am closing the issue. 

https://github.com/SAPDocuments/Tutorials/pull/4423

If you still have questions, feel free to reopen the issue.

**Comment by erikablair** [comment]
Hi, I'm having a similar issue where when trying to do the above go get installs I'm getting the messages:

go get github.com/hyperledger/fabric/core/chaincode/shim: module github.com/hyperledger/fabric@upgrade found (v2.0.0+incompatible), but does not contain package github.com/hyperledger/fabric/core/chaincode/shim

go get github.com/hyperledger/fabric/common/util@v1.4: module github.com/hyperledger/fabric@v1.4 found (v1.4.4), but does not contain package github.com/hyperledger/fabric/common/util

go get github.com/hyperledger/fabric/protos/peer: module github.com/hyperledger/fabric@upgrade found (v2.0.0+incompatible), but does not contain package github.com/hyperledger/fabric/protos/peer

**Comment by chrisfries** [comment]
Hi Erika,

your solution will not work because the **peer** dependency is not the required proto definition. Please use the following package for that `github.com/hyperledger/fabric-protos-go/peer`. 

Then use `go build --tags nopkcs11` to see if it compiles.

However, you will have to vendor those dependencies (`go mod vendor` ) when using the SAP Hyperledger Fabric service. For hyperledger fabric 2.0.0 the chaincode shim and fabric protos where moved to extra repositories. SAP's Hyperledger Fabric service is running on version **1.4.4** and has those new repositories not available during the instantiation of the chaincode.

This is the reason why we came up with the solution to initialize go modules and to get the modules of version 1.4 which is not possible with go get if you don't have go modules initalized.
In the error messages you specified I can see that you run go get without specifying the version. This will fail because in the latest version (2.0.0) the packages were moved.

To sum up you have two options: 
- Get the dependencies in version 1.4 (see Tom's post)
- Use the new packages and vendor the dependencies
